src/displayUI.js

```diff
@@ -0,0 +1,164 @@
+    e.preventDefault();
+    changeColor(param_svg);
+  })
+
+  let param_api = document.createElement("p");
+  param_api.className = "display-data-p";
+  param_api.textContent = `API: ${link.API}`;
+
+  let param_category = document.createElement("p");
+  param_category.className = "display-data-p";
+  param_category.textContent = `Category: ${link.Category}`;
+
+  let param_description = document.createElement("p");
+  param_description.className = "display-data-p";
+  param_description.textContent = `Description: ${link.Description}`;
+
+  let param_link = document.createElement("a");
+  param_link.className = "display-data-p";
+  param_link.href = link.Link;
+  param_link.target = "_blank";
+  param_link.textContent = "Link to an API webpage";
+
+  individuals.append(
+    param_svg,
+    param_api,
+    param_category,
+    param_category,
+    param_description,
+    param_link
+  );
+  document.querySelector("#oneCategoryDisplayBox").append(individuals);
+}
+
+// Display of all Link elements
+export function displayLinksForChosenCategory(arrayOfLinks) {
+  arrayOfLinks.forEach(link => {
+    displayOneLink(link);
+  })
+}
+
+// Change of color for Mark-As-Important feature
+function changeColor(el) {
+  el.classList.toggle('mark-important');
+}
+
+//Search through Categories and display them
+export function searchCategory() {
+  const searchBar = document.querySelector("#searchBar");
+
+  displayAllCategories();
+
+  searchBar.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#categories").innerHTML = "";
+
+    instanceOfMainClass.searchCategories(e).forEach(el => {
+      displayOneCategory(el);
+    })
+  });
+}
+
+// Search through Links and display them
+export function searchLinks() {
+  const searchBarLinks = document.querySelector("#searchBarLinks");
+
+  searchBarLinks.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+
+    instanceOfMainClass.searchAllLinks(e).forEach(el => {
+      displayOneLink(el);
+    })
+  });
+}
+
+export function searchChosenLinksOnly() {
+  const searchBarLinks = document.querySelector("#searchBarLinks");
+
+  searchBarLinks.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+
+    instanceOfMainClass.searchChosenLinks(e).forEach(el => {
+      displayOneLink(el);
+    })
+  });
+}
+
+//btnClear for Category search
+export function btnClear() {
+  let searchContent = document.querySelector("#searchBar");
+  searchContent.value = "";
+  document.querySelector("#categories").innerHTML = "";
+  displayAllCategories();
+}
+
+//btnClear for Links search
+export function btnClearLinks() {
+  let searchContent = document.querySelector("#searchBarLinks");
+  searchContent.value = "";
+  document.querySelector("#oneCategoryDisplayBox").innerHTML = "";
+  instanceOfMainClass.chosenLinks = [];
```

Comment:
What is this: instanceOfMainClass.chosenLinks = [];
Folow encapsulation implment method for changing internal class state, avoid direct outer update of class props.